### DISCUSSION TITLE
Suggestion - Cycle through quests for a POI

### DISCUSSION THREAD

**Discussion by Motorblade** [streetcomplete/StreetComplete#4009] [Ideas]
Often times while surveying, I'll be able to answer one type of quest, but because of the quest order I can't add that information until I solve the higher order quest. 

This leads to situations where people can't add information that they survey due to not being close enough to read things like open times. Adding a way to "cycle" through the unsolved quests without needing to answer the prior ones (but still be able to answer them later) would be very helpful for added whatever information one can.

**Comment by FloEdelmann** [comment]
Two workarounds for this problem:

1. You can open the higher priority quest, tap "Can't say..." -> "Hide".
2. You can change the quest order in the app settings. You could even create several different presets with different active quest types or a different quest order for different kinds of situations.

**Reply by Motorblade** [reply]
Yes, but those aren't perfect solutions. For number one, there isn't really a way to go back and answer those quests later since they disappear completely. The can't say -> hide function is more for quests that you know you cannot reasonably answer, now or in the future.

The second workaround is closer to solving the problem, but it is really cumbersome to have to navigate to the quest order menu every time you want to answer a certain type of quest. It's especially a hassle when you're actively moving and completing quests as a side activity; you don't really have time to shuffle around quests in the order menu just to check for something you can add on the way. Not to mention the quest order menu is relatively hidden and it's usefulness is not immediately obvious to new users.

**Reply by 10992-osm** [reply]
> Yes, but those aren't perfect solutions. For number one, there isn't really a way to go back and answer those quests later since they disappear completely. The can't say -> hide function is more for quests that you know you cannot reasonably answer, now or in the future.

There are, in fact, two ways to do this:

1. Once you have hidden a quest and you want to unhide it, press the undo button, and look for the relevant quest (it will have the quest's icon with a crossed-out eye overlaid).  Click that quest, and confirm undo.
2. Or, more generally, you can unhide all quests you have ever hidden, by going into settings (from the burger menu), scrolling down and pressing "Restore hidden quests".

> The second workaround is closer to solving the problem, but it is really cumbersome to have to navigate to the quest order menu every time you want to answer a certain type of quest. It's especially a hassle when you're actively moving and completing quests as a side activity; you don't really have time to shuffle around quests in the order menu just to check for something you can add on the way. Not to mention the quest order menu is relatively hidden and it's usefulness is not immediately obvious to new users.

---

See also:

- #124
